src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -376,6 +376,15 @@ private void OnPreviewMouseLeftButtonDown(object sender, MouseButtonEventArgs e)
         {
             nodeWasClicked = true;
             BringToFront();
+            if (e.ClickCount == 2)
+            {
+                Debug.WriteLine("Name double clicked!");
+                if (ViewModel != null && ViewModel.RenameCommand.CanExecute(null))
+                {
+                    ViewModel.RenameCommand.Execute(null);
+                }
+                e.Handled = true;
+            }
```

Comment:
The part is not clear to me is that for nodes in other state, how are we handling the double click? On a different route?

Reply:
Should Python script nodes with preview disabled also be editable? This code only accounts for rename, but the code editor of those nodes can be opened as well by double clicking. It actually depends on the part of the node where you double click.

Reply:
Just confirmed with Sol, nodes should/can be renamed in any state.

Reply:
But isn't this change overriding the behaviour of the handler, by handling the double click as a rename and marking the event as handled? Can you check that you can still edit a Python node by double clicking after your change?

Reply:
Yup, @zeusongit if you can confirm the other cases, I am fine with this change

Reply:
Can confirm.
![python preview](https://user-images.githubusercontent.com/32665108/86835151-b2acf580-c069-11ea-9a91-0ebb5c86d65e.gif)